docs/auth.md

```diff
@@ -4,35 +4,224 @@ linkTitle: "Authentication"
+to use that credential.
+
+In this example, Tekton uses a `basic-auth` (username/password pair) `Secret` to access
+Git repositories at `github.com` and `gitlab.com` as well as Docker repositories at `gcr.io`:
 
-This document defines how authentication is provided during execution of a
-`TaskRun` or a `PipelineRun` (referred to as `Runs` in this document).
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  annotations:
+    tekton.dev/git-0: https://github.com
+    tekton.dev/git-1: https://gitlab.com
+    tekton.dev/docker-0: https://gcr.io
+type: kubernetes.io/basic-auth
+stringData:
+  username: <cleartext username>
+  password: <cleartext password>
+```
+
+And in this example, Tekton uses an `ssh-auth` `Secret` to access Git repositories
+at `github.com` only:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  annotations:
+    tekton.dev/git-0: github.com
+type: kubernetes.io/ssh-auth
+stringData:
+  ssh-privatekey: <private-key>
+  # This is non-standard, but its use is encouraged to make this more secure.
+  # Omitting this results in the server's public key being blindly accepted.
+```
+
+## Authenticating as a non-root user
+
+In certain scenarios you might need to authenticate as a non-root user. For example:
+
+- Your platform randomizes the user and/or groups that your containers use to execute.
+- The `Steps` in your `Task` define a non-root `securityContext`.
+- Your `Task` specifies a global non-root `securityContext` that applies to all `Steps` in the `Task`.
+
+The following are considerations for executing `Runs` as a non-root user:
+
+- `ssh-auth` for Git requires the user to have a valid home directory configured in `/etc/passwd`.
+  Specifying a UID that has no valid home directory results in authentication failure.
+- Since SSH authentication ignores the `$HOME` environment variable, you must either move or symlink
+  the appropriate `Secret` files from the `$HOME` directory defined by Tekton (`/tekton/home`) to
+  the the non-root user's valid home directory to use SSH authentication for either Git or Docker.
 
-The build system supports two types of authentication, using Kubernetes'
-first-class `Secret` types:
+For an example of configuring SSH authentication in a non-root `securityContext`,
+see [`authenticating-git-commands`](../examples/v1beta1/taskruns/authenticating-git-commands.yaml).
 
-- `kubernetes.io/basic-auth`
-- `kubernetes.io/ssh-auth`
+## Limiting `Secret` access to specific `Steps`
 
-Secrets of these types can be made available to the `Run` by attaching them to
-the `ServiceAccount` as which it runs.
+As described earlier in this document, Tekton stores supported `Secrets` in
+`$HOME/tekton/home` and makes them available to all `Steps` within a `Task`. 
 
-## Exposing credentials
+If you want to limit a `Secret` to only be accessible to specific `Steps` but not
+others, you must explicitly specify a `Volume` within the `Secret` definition and
```

Comment:
Suggest "specify a `Volume` using the `Secret` definition".  The volume isn't defined in the Secret.

Reply:
Reworded.